api-metastore/src/main/java/org/zalando/nakadi/service/validation/EventTypeAnnotationsValidator.java

```diff
@@ -0,0 +1,103 @@
+
+    private final FeatureToggleService featureToggleService;
+    private final AuthorizationService authorizationService;
+    private final List<String> enforcedAuthSubjects;
+
+    @Autowired
+    public EventTypeAnnotationsValidator(
+            final FeatureToggleService featureToggleService,
+            final AuthorizationService authorizationService,
+            @Value("${nakadi.data_lake.annotations.enforced_auth_subjects:}") final List<String> enforcedAuthSubjects
+    ) {
+        this.featureToggleService = featureToggleService;
+        this.authorizationService = authorizationService;
+        this.enforcedAuthSubjects = enforcedAuthSubjects;
+    }
+
+    public void validateAnnotations(final Map<String, String> annotations) throws InvalidEventTypeException {
+        validateDataLakeAnnotations(Optional.ofNullable(annotations).orElseGet(Collections::emptyMap));
+    }
+
+    private void validateDataLakeAnnotations(@NotNull final Map<String, String> annotations) {
+        final var materializeEvents = annotations.get(MATERIALIZE_EVENTS_ANNOTATION);
+        final var retentionPeriod = annotations.get(RETENTION_PERIOD_ANNOTATION);
+
+        if (materializeEvents != null) {
+            if (!materializeEvents.equals("off") && !materializeEvents.equals("on")) {
+                throw new InvalidEventTypeException(
+                        "Annotation " + MATERIALIZE_EVENTS_ANNOTATION
+                        + " is not valid. Provided value: \""
+                        + materializeEvents
+                        + "\". Possible values are: \"on\" or \"off\".");
+            }
+            if (materializeEvents.equals("on")) {
+                if (retentionPeriod == null) {
+                    throw new InvalidEventTypeException("Annotation " + RETENTION_PERIOD_ANNOTATION
+                            + " is required, when " + MATERIALIZE_EVENTS_ANNOTATION + " is \"on\".");
+                }
+            }
+        }
+
+        if (retentionPeriod != null) {
+            final var retentionReason = annotations.get(RETENTION_REASON_ANNOTATION);
+            if (retentionReason == null || retentionReason.isEmpty()) {
+                throw new InvalidEventTypeException(
+                        "Annotation " + RETENTION_REASON_ANNOTATION + " is required, when "
+                        + RETENTION_PERIOD_ANNOTATION + " is specified.");
+            }
+
+            if (!ANNOTATIONS_PERIOD_PATTERN.matcher(retentionPeriod).find()) {
+                throw new InvalidEventTypeException(
+                        "Annotation " + RETENTION_PERIOD_ANNOTATION
+                        + " does not comply with regex. See documentation "
+                        + "(https://docs.google.com/document/d/1-SwwpwUqauc_pXu-743YA1gO8l5_R_Gf4nbYml1ySiI"
+                        + "/edit#heading=h.kmvigbxbn1dj) for more details.");
+            }
+        }
+
+        if (areDataLakeAnnotationsMandatory()) {
```

Comment:
makes sense to check it first otherwise everything is checked and then exception..